py/server/deephaven/table.py

```diff
@@ -1656,3 +1664,849 @@ def partitioned_transform(self, other: PartitionedTable, func: Callable[[Table,
+            :meth:`~PartitionedTable.partitioned_transform`, default is True
+        sanity_check_joins (bool): whether to check that join operations will only find a given join key in
+            one constituent table and the right table argument if it is also a proxy, default is True
+    """
+    j_object_type = _JPartitionedTableProxy
+
+    @property
+    def j_object(self) -> jpy.JType:
+        return self.j_pt_proxy
+
+    def __init__(self, j_pt_proxy):
+        self.j_pt_proxy = jpy.cast(j_pt_proxy, _JPartitionedTableProxy)
+        self.require_matching_keys = self.j_pt_proxy.requiresMatchingKeys()
+        self.sanity_check_joins = self.j_pt_proxy.sanityChecksJoins()
+        self.target = PartitionedTable(j_partitioned_table=self.j_pt_proxy.target())
+
+    def head(self, num_rows: int) -> PartitionedTableProxy:
+        """Applies the :meth:`~Table.head` table operation to all constituent tables of the underlying partitioned
+        table, and produces a new PartitionedTableProxy with the result tables as the constituents of its underlying
+        partitioned table.
+
+        Args:
+            num_rows (int): the number of rows at the head of the constituent tables
+
+        Returns:
+            a new PartitionedTableProxy
+
+        Raises:
+            DHError
+        """
+        try:
+            return PartitionedTableProxy(j_pt_proxy=self.j_pt_proxy.head(num_rows))
+        except Exception as e:
+            raise DHError(e, "head operation on the PartitionedTableProxy failed.") from e
+
+    def tail(self, num_rows: int) -> PartitionedTableProxy:
+        """Applies the :meth:`~Table.tail` table operation to all constituent tables of the underlying partitioned
+        table, and produces a new PartitionedTableProxy with the result tables as the constituents of its underlying
+        partitioned table.
+
+        Args:
+            num_rows (int): the number of rows at the end of the constituent tables
+
+        Returns:
+            a new PartitionedTableProxy
+
+        Raises:
+            DHError
+        """
+        try:
+            return PartitionedTableProxy(j_pt_proxy=self.j_pt_proxy.tail(num_rows))
+        except Exception as e:
+            raise DHError(e, "tail operation on the PartitionedTableProxy failed.") from e
+
+    def reverse(self) -> PartitionedTableProxy:
+        """Applies the :meth:`~Table.reverse` table operation to all constituent tables of the underlying partitioned
+        table, and produces a new PartitionedTableProxy with the result tables as the constituents of its underlying
+        partitioned table.
+
+        Returns:
+            a new PartitionedTableProxy
+
+        Raises:
+            DHError
+        """
+        try:
+            return PartitionedTableProxy(j_pt_proxy=self.j_pt_proxy.reverse())
+        except Exception as e:
+            raise DHError(e, "reverse operation on the PartitionedTableProxy failed.") from e
+
+    @auto_locking_op
```

Comment:
I don't see an `is_refreshing` attribute on `PartitionedTableProxy`. I assume auto-locking does not work right now, except based on `Table` RHSs?